src/celeritas/global/ActionLauncher.device.hh

```diff
@@ -29,16 +32,34 @@ namespace
 {
 //---------------------------------------------------------------------------//
 /*!
- * Launch the given executor up to the maximum thread count.
+ * Launch the given executor using thread ids in the thread_range
  */
 template<class F>
 __global__ void
-launch_action_impl(size_type const num_threads, F execute_thread)
+launch_action_impl(Range<ThreadId> const thread_range, F execute_thread)
 {
     auto tid = celeritas::KernelParamCalculator::thread_id();
-    if (!(tid < num_threads))
+    if (!(tid < thread_range.size()))
         return;
-    execute_thread(tid);
+    execute_thread(*(thread_range.cbegin() + tid.get()));
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Checks that the TrackOrder will sort tracks by actions applied at the given
+ * ActionOrder. This should match the mapping in the \c SortTracksAction
+ * constructor.
+ *
+ * TODO: Have a single source of truth for mapping TrackOrder to ActionOrder
+ */
+bool is_action_sorted(ActionOrder action, TrackOrder track)
+{
+    return (action == ActionOrder::post
+            && track == TrackOrder::sort_step_limit_action)
+           || (action == ActionOrder::along
+               && track == TrackOrder::sort_along_step_action)
+           || (track == TrackOrder::sort_action
+               && (action == ActionOrder::post || action == ActionOrder::along));
```

Comment:
Since we define ActionOrder and TrackOrder in `celeritas/Types.{hh,cc}` let's move this function there: if we add a new track/action order that'll make it easier to find? Later we can define an ActionOrder-to-TrackOrder mapping and implement the function in terms of that.